build.gradle

```diff
@@ -1,116 +1,70 @@
 plugins {
-    id 'java'
-    id 'jacoco'
-    id 'checkstyle'
     id 'eclipse'
     id 'idea'
-    id "com.github.spotbugs" version "1.6.6"
-    id "org.springframework.boot" version "2.1.1.RELEASE"
-    id "io.spring.dependency-management" version "1.0.6.RELEASE"
-    id "com.palantir.docker" version "0.20.1"
-    id "org.sonarqube" version "2.6.2"
-    id "com.gorylenko.gradle-git-properties" version "1.5.2"
+    id "org.sonarqube" version "2.7" apply false
+    id "com.github.spotbugs" version "1.6.9" apply false
 }
 
-
-group = 'com.bnc'
-version = '0.0.1-SNAPSHOT'
-
-sourceCompatibility = 1.8
-targetCompatibility = 1.8
-
-compileJava.options.encoding = "UTF-8"
-
-repositories {
-    mavenCentral()
-    maven { url "https://repo.spring.io/milestone" }
-}
-
-spotbugs {
-    toolVersion = '3.1.10'
-}
-
-jacoco {
-    toolVersion = "0.8.2"
-}
-
-checkstyle {
-    toolVersion '8.15'
-}
-
-springBoot {
-    buildInfo()
-}
-
-tasks.withType(com.github.spotbugs.SpotBugsTask) {
-    reports {
-        xml.enabled = false
-        html.enabled = true
+allprojects {
+    apply plugin: 'java'
+    apply plugin: 'jacoco'
+    apply plugin: 'checkstyle'
+    apply plugin: 'com.github.spotbugs'
+    apply plugin: 'org.sonarqube'
+
+    group = 'com.bnc'
+    version = '0.0.1-SNAPSHOT'
+    sourceCompatibility = 1.11
```

Comment:
Perhaps use https://docs.gradle.org/current/javadoc/org/gradle/api/JavaVersion.html#VERSION_11

Reply:
Modified to use the object now